src/middlewares/userMiddlewares.js

```diff
@@ -18,11 +19,30 @@ export default class UserMiddlewares {
   static async checkUserExists(req, res, next) {
     try {
       const { email } = req.body;
-      const data = await userServices.getUserByEmail(email);
+      const data = await UserServices.getUserByEmail(email);
       if (!data) return next();
       return resError(res, 409, 'Unsuccesful, user already exists, kindly use a different email.');
     } catch (error) {
       return resError(res, 500, error.message);
     }
   }
+
+  /**
+ * @name userVerified
+ * @description Checks if a user exists in the database
+ * @param {object} req The request object
+ * @param {object} res The response object
+ * @param {object} next The response object
+ * @returns {object} The API response or next()
+ */
+  static async isUserVerified(req, res, next) {
+    try {
+      const userDetails = Utils.verifyToken(req.query.token);
+      const user = await UserServices.getUserById(userDetails.id);
+      if (user.isVerified === false) return next();
+      return resError(res, 400, 'Unsuccesful, Your account has been Verified.');
```

Comment:
@misanj You may want to use `Unsuccessful, Your account has already been verified` Instead so the user does not get confused.